src/middleware/mod.rs

```diff
@@ -26,30 +102,42 @@
-//! of the error, representing the action to take in response to the error, is a complete
-//! Response, which will be sent at the end of the error flow.
+//! The cause of the error is represented simply by the error itself, and the
+//! result of the error, representing the action to take in response to the
+//! error, is a complete Response, which will be sent at the end of the error
+//! flow.
 //!
-//! When an error is thrown in Iron by any middleware or handler returning an `Err`
-//! variant with an `IronError`, the flow of the Request switches to the error flow,
-//! which proceeds to just call the `catch` method of middleware and sidesteps the
-//! `Handler` entirely, since there is already a Response in the error.
+//! When an error is thrown in Iron by any middleware or handler returning an
+//! [`Err`] variant with an [`IronError`], the flow of the [`Request`] switches
+//! to the error flow, which proceeds to just call the `catch` method of
+//! middleware and sidesteps the [`Handler`] entirely, since there is already a
+//! [`Response`] in the error.
 //!
-//! A Request can exit the error flow by returning an Ok from any of the catch methods.
-//! This resumes the flow at the middleware immediately following the middleware which
-//! handled the error. It is impossible to "go back" to an earlier middleware that was
-//! skipped.
+//! A [`Request`] can exit the error flow by returning an Ok from any of the
+//! catch methods. This resumes the flow at the middleware immediately following
+//! the middleware which handled the error. It is impossible to "go back" to an
+//! earlier middleware that was skipped.
 //!
-//! Generally speaking, returning a 5XX error code means that the error flow should be
-//! entered by raising an explicit error. Dealing with 4XX errors is trickier, since
-//! the server may not want to recognize an error that is entirely the clients fault;
-//! handling of 4XX error codes is up to to each application and middleware author.
+//! Generally speaking, returning a 5XX error code means that the error flow
+//! should be entered by raising an explicit error. Dealing with 4XX errors is
+//! trickier, since the server may not want to recognize an error that is
+//! entirely the clients fault; handling of 4XX error codes is up to to each
+//! application and middleware author.
 //!
-//! Middleware authors should be cognizant that their middleware may be skipped during
-//! the error flow. Anything that *must* be done to each Request or Response should
-//! be run during both the normal and error flow by implementing the `catch` method to
-//! also do the necessary action.
+//! Middleware authors should be cognizant that their middleware may be skipped
+//! during the error flow. Anything that *must* be done to each [`Request`] or
+//! [`Response`] should be run during both the normal and error flow by
+//! implementing the `catch` method to also do the necessary action.
 //!
+//! [`Handler`]: trait.Handler.html
```

Comment:
Unsure about those kind of links. It might be better to just use normal codeblocks and wait until rustdoc has a proper way to reference types and traits.